flow/render_status.py

```diff
@@ -0,0 +1,211 @@
+            :type scheduler_status_code:
+                Dictionary
+            :param all_ops:
+                Boolean value indicate if all operations should be displayed
+            :type all_ops:
+                Boolean
+            """
+
+            if scheduler_status_code[job_op['scheduler_status']] != 'U' or \
+               job_op['eligible'] or all_ops:
+                return True
+            else:
+                return False
+
+        def get_operation_status(operation_info, symbols):
+            """Determine the status of an operation.
+
+            :param operation_info:
+                Dicionary containing operation information
+            :type operation_info:
+                Dictionary
+            :param symbols:
+                Dicionary containing code for different job status
+            :type symbols:
+                Dictionary
+            """
+
+            if operation_info['scheduler_status'] >= JobStatus.active:
+                op_status = u'running'
+            elif operation_info['scheduler_status'] > JobStatus.inactive:
+                op_status = u'active'
+            elif operation_info['completed']:
+                op_status = u'completed'
+            elif operation_info['eligible']:
+                op_status = u'eligible'
+            else:
+                op_status = u'ineligible'
+
+            return symbols[op_status]
+
+        if pretty:
+            def highlight(s, eligible, prefix_str, suffix_str):
+                """Change font to bold within jinja2 template
+
+                :param s:
+                    The string to be printed
+                :type s:
+                    str
+                :param eligible:
+                    Boolean value for job eligibility
+                :type eligible:
+                    Boolean
+                """
+                if eligible:
+                    return prefix_str + s + suffix_str
+                else:
+                    return s
+        else:
+            def highlight(s, eligible, prefix_str, suffix_str):
+                """Change font to bold within jinja2 template
+
+                :param s:
+                    The string to be printed
+                :type s:
+                    str
+                :param eligible:
+                    Boolean value for job eligibility
+                :type eligible:
+                    boolean
+                """
+                return s
+
+        template_environment.filters['highlight'] = highlight
+        template_environment.filters['draw_progressbar'] = draw_progressbar
+        template_environment.filters['get_operation_status'] = get_operation_status
+        template_environment.filters['job_filter'] = job_filter
+
+        template = template_environment.get_template(template)
+
+        if option == 'terminal':
+            self.generate_terminal_output(template, context)
+            print(self.terminal_output, file=file)
```

Comment:
We should make this consistent; either `render` always writes its output to a provided file, or it always returns it. Right now it's inconsistent because terminal output is both printed and returned, whereas the other outputs are only returned.